sql/drop.go

```diff
@@ -0,0 +1,48 @@
+package sql
+
+import (
+	"fmt"
+
+	"github.com/cockroachdb/cockroach/keys"
+	"github.com/cockroachdb/cockroach/sql/parser"
+	"github.com/cockroachdb/cockroach/structured"
+)
+
+// DropTable drops a table.
+func (p *planner) DropTable(n *parser.DropTable) (planNode, error) {
```

Comment:
If the user wants to keep the data they can rename the table. I think DROP/TRUNCATE to do what they promise.